contracts/test/bridge/tokenBridge.ts

```diff
@@ -0,0 +1,184 @@
+  let newStarkNetERC20Bridge: Contract
+
+  before(async () => {
+    owner = await starknet.deployAccount('OpenZeppelin')
+
+    let tokenBridgeFactory = await starknet.getContractFactory(
+      '../../node_modules/@chainlink-dev/starkgate-contracts/artifacts/token_bridge.cairo',
+    )
+    tokenBridgeContract = await tokenBridgeFactory.deploy({
+      governor_address: owner.starknetContract.address,
+    })
+
+    let linkTokenFactory = await starknet.getContractFactory('link_token')
+    linkTokenContract = await linkTokenFactory.deploy({ owner: tokenBridgeContract.address })
+
+    const accounts = await ethers.getSigners()
+    deployer = accounts[0]
+
+    const contract = await loadStarkgateContract('StarknetERC20Bridge')
+    const starkNetERC20BridgeFactory = new ethers.ContractFactory(
+      contract.abi,
+      contract.bytecode,
+      deployer,
+    )
+    starkNetERC20Bridge = await starkNetERC20BridgeFactory.deploy()
+    await starkNetERC20Bridge.deployed()
+
+    const mockStarknetMessagingFactory = await ethers.getContractFactory(
+      'MockStarkNetMessaging',
+      deployer,
+    )
+    mockStarknetMessaging = await mockStarknetMessagingFactory.deploy()
+    await mockStarknetMessaging.deployed()
+
+    await starknet.devnet.loadL1MessagingContract(networkUrl, mockStarknetMessaging.address)
+
+    const testERC20Factory = await ethers.getContractFactory('TestERC20', deployer)
+    testERC20 = await testERC20Factory.deploy()
+    await testERC20.deployed()
+
+    const proxyContract = loadProxyContract('Proxy')
+    const proxyFactory = new ethers.ContractFactory(
+      proxyContract.abi,
+      proxyContract.bytecode,
+      deployer,
+    )
+    proxy = await proxyFactory.deploy(0)
+    await proxy.deployed()
+  })
+
+  describe('Test bridge from L1 to L2', function () {
+    it('Test Set and Get function for L2 token address', async () => {
+      const new_data = ethers.utils.hexConcat([
+        ethers.utils.hexZeroPad(ethers.constants.AddressZero, 32),
+        ethers.utils.hexZeroPad(testERC20.address, 32),
+        ethers.utils.hexZeroPad(mockStarknetMessaging.address, 32),
+      ])
+
+      await proxy.connect(deployer).addImplementation(starkNetERC20Bridge.address, new_data, false)
+      await proxy.connect(deployer).upgradeTo(starkNetERC20Bridge.address, new_data, false)
+    })
+
+    it('Should add implementation and upgrade successfully', async () => {
+      await owner.invoke(tokenBridgeContract, 'set_l2_token', {
+        l2_token_address: linkTokenContract.address,
+      })
+      const { res: l2_address } = await tokenBridgeContract.call('get_l2_token', {})
+      expectAddressEquality(l2_address.toString(), linkTokenContract.address)
+    })
+
+    it('Should wrap contract and set L2 TokenBridge successfully', async () => {
+      const contract = await loadStarkgateContract('StarknetERC20Bridge')
+      newStarkNetERC20Bridge = await ethers.getContractAt(contract.abi, proxy.address)
```

Comment:
Please define this at the beginning of the test, as part of the before hook or smth. It's used in multiple tests, so it makes sense to set it at the beginning - not just randomly in one of the tests first time you need to use it.